### PR TITLE
Add a script for porting passes tests to lit

### PR DESCRIPTION
**Pull request by tlively**
And use it to port the very simple untee test.

### CODE REVIEW

File: scripts/port_test.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('-f', '--force', action='store_true',
+                        help='Overwrite existing lit tests')
+    parser.add_argument('--no-delete', action='store_true',
+                        help='Do not remove the old tests')
+    parser.add_argument('--git-add', action='store_true',
+                        help='Stage changes')
+    parser.add_argument('tests', nargs='+', help='The test files to port')
+    args = parser.parse_args()
```
Comment (aheejin) [inline scripts/port_test.py:96]: It'd be good to add simple `--help` with a usage string

File: scripts/port_test.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    passes_file = os.path.join(test_dir, 'passes', base + '.passes')
+    if os.path.exists(passes_file):
+        with open(passes_file) as f:
+            joined_passes = f.read().strip()
+
+    passes = joined_passes.split('_')
+    opts = [('--' + p if not p.startswith('O') and p != 'g' else '-' + p)
+            for p in passes]
+
+    run_line = (f';; RUN: foreach %s %t wasm-opt {" ".join(opts)} -S -o -'
```
Comment (aheejin) [inline scripts/port_test.py:66]: Just saw we had `foreach` script: https://github.com/WebAssembly/binaryen/commit/8de4349ebdb6ce398ef4939f9f375a7413d99b09

Was this added in part of a PR? Can't see any link to it..

File: scripts/port_test.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          print(src_file.read(), file=dest_file, end='')
+
+    update_script = os.path.join(script_dir, 'update_lit_checks.py')
+    subprocess.run([sys.executable, update_script, '-f', '--all-items', dest])
+
+    if not args.no_delete:
+        for f in glob.glob(test.replace('.wast', '.*')):
+            os.remove(f)
+            if args.git_add:
+                subprocess.run(['git', 'add', f])
```
Comment (aheejin) [inline scripts/port_test.py:81]: Why `git add` the source file?

**Review by aheejin — COMMENTED** [review]
Nice!

**Review by aheejin — APPROVED** [review]

File: scripts/port_test.py (hunk 1 above)
Comment (tlively) [reply scripts/port_test.py:96]: This is automatically generated by the arg parser module!

File: scripts/port_test.py (hunk 2 above)
Comment (tlively) [reply scripts/port_test.py:66]: Yes, it was hidden in https://github.com/WebAssembly/binaryen/pull/3939

File: scripts/port_test.py (hunk 3 above)
Comment (tlively) [reply scripts/port_test.py:81]: This stages the *removal* of the file.

**Comment by tlively** [comment]
I fixed the flake8 error and added a new note to be emitted into the ported test identifying it as automatically ported. I realized this actually depends on https://github.com/WebAssembly/binaryen/pull/3962, so I will wait for that to land before re-generating the test here and landing this.

File: scripts/port_test.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
```
Comment (kripken) [inline scripts/port_test.py:1]: maybe `port_pass_test_to_lit.py`?

File: scripts/port_test.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    subprocess.run([sys.executable, update_script, '-f', '--all-items', dest])
+
+    if not args.no_delete:
+        for f in glob.glob(test.replace('.wast', '.*')):
+            os.remove(f)
+            if args.git_add:
+                subprocess.run(['git', 'add', f])
+
+    if args.git_add:
+        subprocess.run(['git', 'add', dest])
```
Comment (kripken) [inline scripts/port_test.py:90]: I'm surprised to see the git operations here... I'd expect the user to run them manually. But I don't feel strongly, if this is easier sgtm.

**Review by kripken — APPROVED** [review]

File: scripts/port_test.py (hunk 5 above)
Comment (tlively) [reply scripts/port_test.py:90]: Yep, it's a little weird to do this in the script, so I put it behind a flag. It's a great quality of life feature, though.

**Pull request merged**